lib/file/bloc.dart

```diff
@@ -97,4 +102,92 @@ class FileBloc {
       throw PermissionNotGranted();
     }
   }
+
+  Future<void> uploadFile({
+    @required BuildContext context,
+    @required Id<dynamic> owner,
+    Id<File> parent,
+  }) async {
+    assert(context != null);
+    assert(owner != null);
+
+    // Let the user pick files.
+    final files = await FilePicker.getMultiFile();
+
+    for (var i = 0; i < files.length; i++) {
+      final file = files[i];
+
+      final snackbar = Scaffold.of(context).showSnackBar(SnackBar(
+        duration: Duration(days: 1),
+        content: Row(
+          children: <Widget>[
+            Transform.scale(
+              scale: 0.5,
+              child: CircularProgressIndicator(
+                valueColor: AlwaysStoppedAnimation(Colors.white),
+              ),
+            ),
+            SizedBox(width: 8),
+            Expanded(
+              child: Text(
+                files.length == 1
+                    ? 'Uploading ${file.name}'
+                    : 'Uploading ${file.name} (${i + 1} / ${files.length})',
+              ),
+            ),
+          ],
+        ),
+      ));
+
+      await _uploadSingleFile(file: file, owner: owner, parent: parent);
+
+      snackbar.close();
```

Comment:
I really don't like this mix of widgets and logic. Suggestion: Return a stream with the necessary information to update the snack bar which is created somewhere in a widget file. (Note: consume the stream inside the snack bar so when the user dismisses it, they don't get new snack bars repeatedly, except the 'Done' message.) This is also missing L10n and a final snack bar when the upload is completed.

Reply:
Maybe include a button in the final snack bar to open the folder uploaded to? And we should notify the user where they are uploading to beforehand.

Reply:
I believe we manage to upload to the current folder in MVP MVP, so I won't add a localized string notifying the user about the upload folder.
The button should probably be put into another issue.